src/js/components/Meter.js

```diff
@@ -683,6 +692,134 @@ var Meter = React.createClass({
+      endangle = this._translateEndAngle(angle, indicator);
+      start = this._translateEndAngle(angle, indicator - 1);
+      radio = this.state.startRadius;
+      commands = this._spiralCommands(start, endangle, radio);
+    }
+    return this._buildPath(commands, null, null, index, classes);
+  },
+
+  _maxValue: function() {
+    //Get high value for high indicator
+    var high = this._seriesMax(this.state.series);
+    if (high > this.state.highIndicator) {
+      this.state.highIndicator = high;
+      this.state.date1 = this._createDate("/");
+    }
+  },
+
+  _sumValues:function(e) {
+    //Acumulated values for average values
+    var sum = 0;
+    var self = this;
+    e.map(function(data, i) {
+      sum += data.value;
+      self.state.averageCounter += 1;
+    });
+    return sum;
+  },
+
+  _averageValue: function() {
+    //Get average value for average indicator
+    if (this.state.mouse === false) {
+      this.state.averageSum += this._sumValues(this.state.series);
+      this.state.averageIndicator = this.state.averageSum / this.state.averageCounter;
+      this.state.date2 = this._createDate("/");
+    }
+  },
+
+  _createDate: function(c){
+    //Make custom date for high and average indicators
+    var date = new Date();
+    var day = date.getDate();
+    if (date.getDate()<10)
+    day="0"+date.getDate();
+    var month = date.getMonth()+1;
+    if (date.getMonth()+1<10)
+    month="0"+date.getMonth()+1;
+    var year = date.getFullYear();
+    var hour = date.getHours();
+    if (date.getHours()<10)
+    hour="0"+date.getHours();
+    var mins = date.getMinutes();
+    if (date.getMinutes()<10)
+    mins="0"+date.getMinutes();
+    var secs = date.getSeconds();
+    if (date.getSeconds()<10)
+    secs="0"+date.getSeconds();
+    var fech;
+    if (c=="-") fech = month+"-"+day+"-"+year+" "+hour+":"+mins+":"+secs;
+    if (c=="/") fech = month+"/"+day+"/"+year+" "+hour+":"+mins+":"+secs;
+    return fech;
+  },
+
+   _handleMouseOver: function(i) {
+    //handle mouse over for indicators to show tooltip   
+    var message;
+    if (i.dispatchMarker === ".1.0.0.0.0.2.3.$1") {
+      this.setState({mouseY: i.clientY});
+      this.setState({mouseX: i.clientX});    
+      message = "-Value: " + this.state.highIndicator.toFixed(2) + " -Date:" + this.state.date1;
+      this.setState({message: message});
+      this.setState({showTooltip: true});
+    }
+    if (i.dispatchMarker === ".1.0.0.0.0.2.4.$2") {
+      this.setState({mouseY: i.clientY});
+      this.setState({mouseX: i.clientX});
+      message = "-Value: " + this.state.averageIndicator.toFixed(2) + " -Date:" + this.state.date2;
+      this.setState({message: message});
+      this.setState({showTooltip: true});
+    }
+    this.setState({mouse: true});
+  },
+
+  _handleMouseOut: function(i) {
+    //handle mouse out for indicators to hide tooltip
+    this.setState({message: ""});
```

Comment:
I don't think we need to call setState all the time. in general this would be sufficient:

`this.setState({message: '', showTooltip: false, mouse: false})`